tests/test-file-cache.php

```diff
@@ -102,4 +102,104 @@ public function test_export() {
 	}
+
+	public function test_import() {
+		$max_size  = 32;
+		$ttl       = 60;
+		$cache_dir = Utils\get_temp_dir() . uniqid( 'wp-cli-test-file-cache', true );
+		$cache     = new FileCache( $cache_dir, $ttl, $max_size );
+
+		$tmp_dir = Utils\get_temp_dir() . uniqid( 'wp-cli-test-file-cache-import', true );
+		mkdir( $tmp_dir );
+
+		// "$group/$slug-$version.$ext";
+		$key              = 'plugin/my-fixture-plugin-1.0.0.zip';
+		$fixture_filepath = $tmp_dir . '/my-downloaded-fixture-plugin-1.0.0.zip';
+
+		$zip = new ZipArchive();
+		$zip->open( $fixture_filepath, ZIPARCHIVE::CREATE );
+		$zip->addFile( __FILE__ );
+		$zip->close();
+
+		$result = $cache->import( $key, $fixture_filepath );
+
+		// Assert file is imported.
+		self::assertTrue( $result );
+		self::assertFileExists( "{$cache_dir}/{$key}" );
+
+		// Clean up.
+		$cache->clear();
+		unlink( $fixture_filepath );
+		rmdir( $tmp_dir );
+	}
+
+	/**
+	 * @see https://github.com/wp-cli/wp-cli/pull/5947
+	 */
+	public function test_import_do_not_use_cache_file_cannot_be_read() {
+		$max_size  = 32;
+		$ttl       = 60;
+		$cache_dir = Utils\get_temp_dir() . uniqid( 'wp-cli-test-file-cache', true );
+		$cache     = new FileCache( $cache_dir, $ttl, $max_size );
+
+		$tmp_dir = Utils\get_temp_dir() . uniqid( 'wp-cli-test-file-cache-import', true );
+		mkdir( $tmp_dir );
+
+		$key              = 'plugin/my-fixture-plugin-1.0.0.zip';
+		$fixture_filepath = $tmp_dir . '/my-bad-permissions-fixture-plugin-1.0.0.zip';
+
+		$zip = new ZipArchive();
+		$zip->open( $fixture_filepath, ZIPARCHIVE::CREATE );
+		$zip->addFile( __FILE__ );
+		$zip->close();
+
+		chmod( $fixture_filepath, 0000 );
+
+		// "Warning: copy(-.): Failed to open stream: Permission denied".
+		$error = null;
+		set_error_handler(
+			function ( $errno, $errstr ) use ( &$error ) {
+				$error = $errstr;
+			}
+		);
+
+		$result = $cache->import( $key, $fixture_filepath );
+
+		restore_error_handler();
+
+		self::assertNull( $error );
+		self::assertFalse( $result );
+
+		// Clean up.
+		$cache->clear();
+		unlink( $fixture_filepath );
+		rmdir( $tmp_dir );
+	}
+
+	/**
+	 * Windows filenames cannot end in periods.
+	 *
+	 * @covers \WP_CLI\FileCache::validate_key
+	 *
+	 * @see https://github.com/wp-cli/wp-cli/pull/5947
+	 * @see https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions
+	 */
+	public function test_validate_key_ending_in_period() {
+		$max_size  = 32;
+		$ttl       = 60;
+		$cache_dir = Utils\get_temp_dir() . uniqid( 'wp-cli-test-file-cache', true );
+		$cache     = new FileCache( $cache_dir, $ttl, $max_size );
+
+		$key = 'plugin/advanced-sidebar-menu-pro-9.5.7.';
+
+		$reflection = new ReflectionClass( $cache );
+
+		$method = $reflection->getMethod( 'validate_key' );
+		$method->setAccessible( true );
+
+		$result = $method->invoke( $cache, $key );
+
+		self::assertFalse( str_ends_with( $result, '.' ) );
+		self::assertEquals( 'plugin/advanced-sidebar-menu-pro-9.5.7', $result );
```

Comment:
`str_ends_with` does not exist on PHP 5.6+

```suggestion
		$this->assertStringEndsNotWith( '.', $result );
		$this->assertSame( 'plugin/advanced-sidebar-menu-pro-9.5.7', $result );
```